### PR TITLE
Impro 287 reduce warnings in unittests

### PR DESCRIPTION
**Pull request by fionaRust**
Reference issue if exists

Testing:
 - [ ] Ran tests and they passed OK
 - [ ] Added new tests for the new feature(s)

### CODE REVIEW

File: lib/improver/nbhood/vicinity.py [hunk 1]
```diff
@@ -108,6 +109,8 @@ def __repr__(self):
             self.distance, self.neighbourhood_method, self.radii,
             self.lead_times, self.weighted_mode, self.ens_factor)
 
+    @ManageWarnings(
+        ignored_messages=["Collapsing a non-contiguous coordinate."])
```
Comment (benfitzpatrick) [inline lib/improver/nbhood/vicinity.py]: Do we always want to ignore this warning (filter it generally)? If so we could always also have something under lib/improver/__init__.py

File: lib/improver/utilities/warnings_handler.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+"""
+Classes and functions for managing warnings in the IMPROVER code.
+"""
+import warnings
+import sys
+
+
+class ManageWarnings(object):
+    """
+    A decorator use to manage the warnings that are raised by a function.
```
Comment (benfitzpatrick) [inline lib/improver/utilities/warnings_handler.py]: typo

**Comment by benfitzpatrick** [comment]
I like the decorator approach! I think maybe we could always filter out a couple of warnings as well.

File: lib/improver/utilities/warnings_handler.py (hunk 2 above)
Comment (bayliffe) [reply lib/improver/utilities/warnings_handler.py]: ✅

File: lib/improver/nbhood/vicinity.py (hunk 1 above)
Comment (fionaRust) [reply lib/improver/nbhood/vicinity.py]: I've only put this in when I've checked where it is being raised and I know we are happy to suppress it. This is mostly when the cube being collapsed is not the cube output from the plugin, so it does not matter if the cube's metadata gets a bit messed up. In general this warning is suggesting we have created some messy metadata on the cube, so I think we shouldn't always ignore it without fixing the metadata.

**Comment by bayliffe** [comment]
Replaced with PR https://github.com/metoppv/improver/pull/458.

**Pull request closed without merging**